src/appleseed/renderer/utility/dynamicspectrum.h

```diff
@@ -833,32 +631,16 @@ APPLESEED_FORCE_INLINE DynamicSpectrum<float, 31>& operator+=(DynamicSpectrum<fl
 template <typename T, size_t N>
 inline DynamicSpectrum<T, N>& operator-=(DynamicSpectrum<T, N>& lhs, const DynamicSpectrum<T, N>& rhs)
 {
-    assert(lhs.get_intent() == rhs.get_intent());
-
-    if (lhs.size() <= rhs.size())
-    {
-        if (lhs.size() < rhs.size())
-            DynamicSpectrum<T, N>::upgrade(lhs, lhs);
-
-        for (size_t i = 0, e = lhs.size(); i < e; ++i)
-            lhs[i] -= rhs[i];
-    }
-    else
-    {
-        DynamicSpectrum<T, N> up_rhs;
-        DynamicSpectrum<T, N>::upgrade(rhs, up_rhs);
-
-        for (size_t i = 0; i < N; ++i)
-            lhs[i] -= up_rhs[i];
-    }
+    for (size_t i = 0; i < DynamicSpectrum<T, N>::size(); ++i)
+        lhs[i] -= rhs[i];
```

Comment:
Most arithmetic operations are not used (at least not in performance-sensitive contexts) so they haven't been optimized yet.